plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -305,26 +305,33 @@ protected void renameTable(ConnectorSession session, Connection connection, Stri
         if (!remoteSchemaName.equals(newRemoteSchemaName)) {
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
         }
-
-        execute(connection, format(
-                "sp_rename %s, %s",
-                singleQuote(catalogName, remoteSchemaName, remoteTableName),
-                singleQuote(newRemoteTableName)));
+        String fullTableFromName = DOT_JOINER.join(
+                quoted(catalogName),
+                quoted(remoteSchemaName),
+                quoted(remoteTableName));
+
+        try (CallableStatement renameTable = connection.prepareCall("exec sp_rename ?, ?")) {
+            renameTable.setString(1, fullTableFromName);
+            renameTable.setString(2, newRemoteTableName);
+            renameTable.execute();
+        }
     }
 
     @Override
     protected void renameColumn(ConnectorSession session, Connection connection, RemoteTableName remoteTableName, String remoteColumnName, String newRemoteColumnName)
             throws SQLException
     {
-        execute(connection, format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(remoteColumnName) + "]"),
-                "[" + newRemoteColumnName + "]"));
-    }
-
-    private static String escape(String name)
-    {
-        return name.replace("'", "''");
+        String columnFrom = DOT_JOINER.join(
```

Comment:
might not be obvious later that while the `columnFrom` needs quoting the `newRemoteColumnName` does not (even though we use prepared statements). Add a comment explaining why.

Reply:
Might just be:

```
// sp_rename treats first argument as SQL object name so it needs to be properly quoted and escaped. The second argument is treated literally.
```

Reply:
done